powersimdata/data_access/data_access.py

```diff
@@ -35,6 +47,34 @@ def move_to(self, file_name, to_dir, change_name_to=None):
         """
         raise NotImplementedError
 
+    def get_base_dir(self, kind, backup=False):
+        """Get path to given kind relative to instance root
+
+        :param str kind: one of {input, output, tmp}
+        :param bool backup: pass True if relative to backup root dir
+        :raises ValueError: if kind is invalid
+        :return: (*str*) -- the specified path
+        """
+        _allowed = list(_dirs.keys())
+        if kind not in _allowed:
+            raise ValueError(f"Invalid 'kind', must be one of {_allowed}")
+
+        root = self.root if not backup else self.backup_root
+        return self.join(root, *_dirs[kind])
```

Comment:
Not sure I understand what this does with `self.join` being `None`

Reply:
We set `self.join` in the subclasses - assumption is we never work with base class instances, but we could probably enforce that

Reply:
I don't know why I forgot about the child classes ... Sorry about that.

Reply:
`raise NotImplementedError` in the parent?

Reply:
This method is already in the base class, so we would have to do something like set `self.join` to a function that raises the error. Currently I think we would get an `AttributeError` but figure it's ok since nothing would work anyway given a pure `DataAccess` instance.

Reply:
It looks like that `LocalDataAccess` and `SSHDataAccess` don't have much in common anymore.

Reply:
True, mainly they share an interface, but not much implementation. That may be a good thing though - there is a bit more code but it's also more straightforward.